index.html

```diff
@@ -1823,29 +1823,350 @@ <h5><code>invokeaction</code></h5>
+            </thead>
+            <tbody>
+              <tr class="rfc2119-table-assertion">
+                <td><code>status</code></td>
+                <td>The status of the action request.</td>
+                <td>mandatory</td>
+                <td>
+                  <a href="http://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/#string">
+                  <code>string</code></a> (one of <code>pending</code>, 
+                  <code>running</code>, <code>completed</code> or 
+                  <code>failed</code>)
+                </td>
+              </tr>
+              <tr class="rfc2119-table-assertion">
+                <td><code>output</code></td>
+                <td>
+                  The output data, if any, of a completed action which
+                  MUST conform with the <code>output</code> data schema of the 
+                  corresponding 
+                  <a href="https://www.w3.org/TR/wot-thing-description/#actionaffordance">
+                  <code>ActionAffordance</code></a>.
+                </td>
+                <td>optional</td>
+                <td>any type</td>
+              </tr>
+              <tr class="rfc2119-table-assertion">
+                <td><code>error</code></td>
+                <td>
+                  An error message, if any, associated with a failed action 
+                  which MUST conform with the Problem Details format
+                  [[RFC7807]].
+                </td>
+                <td>optional</td>
+                <td><code>object</code></td>
+              </tr>
+            </tbody>
+          </table>
+
+          <h6 id="sync-action-response">Synchronous Action Response</h6>
+          <p class="rfc2119-assertion" 
+          id="profile-5.2.2.1-thing-protocol-binding-actions-invokeaction-7">
+            If providing a Synchronous Action Response, a Web Thing MUST send 
+            an HTTP response with:
+          </p>
+          <ul>
+            <li>Status code set to <code>200</code></li>
+            <li><code>Content-Type</code> header set to 
+              <code>application/json</code></li>
+            <li>A body containing an <a href="#ActionStatus">
+              <code>ActionStatus</code></a> object serialized
+              in JSON</li>
+          </ul>
+          <pre class="example">
+            HTTP/1.1 200 OK
+            Content-Type: application/json
+            {
+              "status": "completed"
+            }
+          </pre>
+          <p class="rfc2119-assertion" 
+          id="profile-5.2.2.1-thing-protocol-binding-actions-invokeaction-8">
+            If a Web Thing encounters an error in attempting to execute an
+            action before sending an Synchronous Action Response, it MUST send 
+            an HTTP error response with:
+          </p>
+          <ul>
+            <li>An error response status code (e.g. <code>400</code> for an 
+              invalid input or <code>500</code> for a failed actuation)</li>
```

Comment:
So one possible error here is caused by devices that cannot support parallel actions of the same type, when a action of that type is currently in progress.  This situation perhaps should have its own error code, so that the consumer knows it can try again later (that the error is temporary).  This situation should be quite common in IoT devices, unlike web services.  It might also be useful to have a flag on actions (e.g. "parallel" : true) but that is something we should propose at the TD level.

Reply:
Note that all operations can return an error response from those recommended in section 5.2.4 Error Responses. If you think there's a particular error code that should be recommended here, could you please file an issue for that?